src/hypothesis/internal/conjecture/utils.py

```diff
@@ -71,8 +71,21 @@ def centered_integer_range(data, lower, upper, center):
     )
 
 
+def is_deprecated_sample_type(values):
+    sequence_types = (OrderedDict, Sequence, enum.EnumMeta)
+    try:
+        from numpy import ndarray
+        if isinstance(values, ndarray) and values.ndim != 1:
+            return True
+        sequence_types += (ndarray,)
```

Comment:
I’d like a comment here explaining why we can only sample 1-dimensional arrays.

Also, I might simplify this code not to mutate `sequence_types`. In particular:

```python
try:
    from numpy import ndarray
    if isinstance(values, ndarray):
        return values.ndim == 1
except ImportError:  # pragma: no cover
    pass
```